tests/integration/metrics_tests/test_metrics.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import time
+import yaml
+
+import ops
+import pytest
+import urllib3
+from pytest_operator.plugin import OpsTest
+from pathlib import Path
+from ..ha_tests import helpers as ha_helpers
+
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+DATABASE_APP_NAME = "mongodb-k8s"
+MONGODB_EXPORTER_PORT = 9216
+MEDIAN_REELECTION_TIME = 12
+
+
+@pytest.fixture(scope="module")
+def chaos_mesh(ops_test: OpsTest) -> None:
+    ha_helpers.deploy_chaos_mesh(ops_test.model.info.name)
+
+    yield
+
+    ha_helpers.destroy_chaos_mesh(ops_test.model.info.name)
+
+
+async def get_address(ops_test: OpsTest, app_name=DATABASE_APP_NAME, unit_num=0) -> str:
+    """Get the address for a unit."""
+    status = await ops_test.model.get_status()  # noqa: F821
+    address = status["applications"][app_name]["units"][f"{app_name}/{unit_num}"]["address"]
+    return address
+
+
+async def verify_endpoints(ops_test: OpsTest, unit: ops.model.Unit) -> str:
+    """Verifies mongodb endpoint is functional on a given unit."""
+    http = urllib3.PoolManager()
+
+    unit_address = await get_address(ops_test=ops_test)
+    mongodb_exporter_url = f"http://{unit_address}:{MONGODB_EXPORTER_PORT}/metrics"
```

Comment:
Maybe we can inline this variable?

Reply:
I think this would make things a little less readable. Is this a necessary change for you? If so I can make it